addons/point_of_sale/static/src/app/models/data_service.js

```diff
@@ -422,6 +422,16 @@ export class PosData extends Reactive {
         return await this.execute({ type: "read", model, ids, fields, options, queue });
     }
 
+    async readNew(model, fields = [], options = [], queue = false) {
+        return await this.searchRead(
+            model,
+            [["id", "not in", this.models[model].getAll().map((record) => record.id)]],
+            fields,
+            options,
+            queue
+        );
+    }
```

Comment:
You can use `read` directly, because when you call up this method you're supposed to know whether you already have the record or not. In the worst case, the existing record will be overwritten.

Reply:
i also dislike this method and i would like to remove it. My problem is that i want to fetch a record but i don't know it's id. In this case i want to fetch the partners that were created in the backend since the pos session was opened

Reply:
In this case, you can retrieve the data with a normal search (searchRead) just after creation with your new form view. Normally, if you search by id desc, you'll get the one you're looking for.

You can use a limit of 1, but if you're worried that someone else has created one at the same time in the backend, use a higher limit.